src/part-guide/concurrency.md

```diff
@@ -31,61 +31,61 @@ Even where IO[^io-def] happens (printing using `println!` is IO - it is outputti
+A thread can also choose to pause itself by calling a [`sleep`](https://doc.rust-lang.org/stable/std/thread/fn.sleep.html) function, usually with a timeout. In this case the OS pauses the thread at the thread's own request. Similar to pausing due to pre-emption or IO, the OS will wake the thread up again later (after the timeout) to continue execution.
 
-When an OS pauses one thread and starts another (for any reason), it is called *context switching*. The context being switched includes the registers, operating system records, and the contents of many caches. That's a non-trivial amount of work. Together with the transfer of control to the OS and back to a thread, and the costs of working with stale caches, context switching is an expensive operation.
+When an OS pauses one thread and starts another (for any reason), it is called *context switching*. The context being switched includes the registers, operating system records, and the contents of many caches. That's a non-trivial amount of work. On top of that, there is a transfer of control to the OS and back to the thread and the costs of working with stale caches. Adding all these costs together makes context switching an expensive operation.
 
-Finally, note that some hardware or OSs do not support processes or threads, this is more likely in the embedded world.
+Finally, note that some hardware or OSs do not support processes or threads. In the embedded world, there is no operating system, so there is nothing to provide processes or threads.[^rtos]
 
 [^proc-program]: from the user's perspective, a single program may include multiple processes, but from the OS's perspective each process is a separate program.
 [^shmem]: Some OSs do support sharing memory between processes, but using it requires special treatment and most memory is not shared.
 [^sched]: Exactly how the OS chooses which thread to run and for how long (and on which core), is a key part of scheduling. There are many options, both high-level strategies and options to configure those strategies. Making good choices here is crucial for good performance, but it is complicated and we won't dig into it here.
-[^busywait]: There's another option which is that the thread can *busy wait* by just spinning in a loop until the IO is finished. This is not very efficient since other threads won't get to run and is uncommon in most modern systems. You may come across it in the implementations of locks or in very simple embedded systems.
-
+[^busywait]: There's another option which is that the thread can *busy wait* by just spinning in a loop until the IO is finished. This is not very efficient, both in terms of electrical energy needed by the CPU and throughput since other threads won't get to run. Busy waiting is uncommon in most modern systems, though you may come across it in the implementations of locks or in very simple embedded systems.
```

Comment:
I think busy waiting can be efficient in terms of throughput which is why it is used :-) I suspect it can be more efficient in terms of energy too since it avoids a context switch. I prefer to keep the use of 'efficient' vague here, but if you strongly prefer to clarify, then I guess saying it "sounds inefficient, but..." or "can be inefficient at times but... " might be better?